src/content_scripts/protoots.js

```diff
@@ -215,13 +226,18 @@ async function addProplate(element) {
 	 * @returns {string}
 	 */
 	function getID(element) {
-		const id = element.dataset.id;
+		let id = element.dataset.id;
 		if (!id) {
 			// We don't have a status ID, pronouns might not be in cache
 			warn(
-				"The element passed to addProplate does not have a data-id attribute, although it should have one.",
+				"The element passed to addProplate does not have a data-id attribute, searching for article.",
 				element,
 			);
+			//if we couldn't get an id from the div try the closest article
+			const ArticleElement = element.closest("article");
+			if (ArticleElement) {
+				id = getID(ArticleElement);
+			}
```

Comment:
Hm, could this be simplified to this?

```suggestion
			id = element.closest("article[data-id]")?.dataset.id
```

But hm. If that selector doesn't match anything, the ID is `undefined`. Which is would not align with the documentation of this method.

What do we do right now if both the passed `element` and the closest element do not have an ID?

Reply:
if we're looking at a detailed-status we grab the ID from the URL, otherwise we don't do anything - fetchPronouns() returns null because we didn't pass a data-id

Reply:
ah thx, so we can leave the code as is. :3